packages/jmix-react-web/src/crud/editor/validation/useClientValidation.ts

```diff
@@ -0,0 +1,58 @@
+import { Metadata, isEntityValid, BeanValidationRule, MainStore, useMainStore } from "@haulmont/jmix-react-core";
+import { useCallback, useState } from "react";
+import { JmixServerValidationErrors } from "../../../common/JmixServerValidationErrors";
+import { IntlShape, useIntl } from 'react-intl';
+
+const validationResultToFieldErrors = (
+    invalidFieldRules: Map<string, BeanValidationRule[]>,
+    entityName: string,
+    intl: IntlShape,
+    mainStore: MainStore
+): Map<string, string[]> => {
+    const result = new Map();
+
+    invalidFieldRules.forEach((v, k) => {
+        result.set(k, v.map(x => {
+            const id = `antd.form.beanValidation.${x.name.toLowerCase()}`;
+            const fieldName = mainStore?.messages?.[`${entityName}.${k}`];
+            
+            return intl.formatMessage({ id }, {
+                ...x,
+                name: fieldName
+            });
+        }))
+    })
+
+    return result;
+}
+
+export function useClientValidation(): [
+    (values: any, entityName: string, metadata: Metadata) => boolean,
+    JmixServerValidationErrors | undefined
+] {
+    const [errors, setErrors] = useState<JmixServerValidationErrors>();
+    const intl = useIntl();
+    const mainStore = useMainStore();
+
+    const executeValidation = useCallback(
+        (values: any, entityName: string, metadata: Metadata) => {
+            const { isValid, invalidFieldRules } = isEntityValid(values, entityName, metadata);
+
+            if (!isValid) {
+                setErrors({
+                    fieldErrors: validationResultToFieldErrors(invalidFieldRules, entityName, intl, mainStore)
+                });
+            } else {
+                setErrors(undefined)
+            }
```

Comment:
(personal remark) I think this code сould look better:
```suggestion
          const fieldsErrors = isValid 
            ? undefined
            : validationResultToFieldErrors(invalidFieldRules, entityName, intl, mainStore)
          setErrors({fieldErrors});
```